### DISCUSSION TITLE
need help.

### DISCUSSION THREAD

**Discussion by abd-4fg** [projectdiscovery/nuclei-templates#9169] [General]
Hi ,
i hate to say but it was just an exhaustive whole week trying to convert nuclei regexes that are present in exposures/tokens templates , 
but all in vain , multiple errors .

What type of regular expressions are they ?
which regex engine nuclei uses ?

And most importantly  i want to know : Which regex engines fully supports all the regexes that are in exposures templates ?
I tried to convert these regexes to be compatible with  grep regex engine and PCRE engine , always an error comes stating certain operaters are not supported from regex ,

Kindly help me , this has become much exhausting task for me.

Regards,

**Comment by ehsandeep** [comment]
@abd-4fg golang uses regex re2 : https://pkg.go.dev/regexp
